store/storage/sqlite/db_test.go

```diff
@@ -87,3 +89,76 @@ func TestDatabase_ReverseIterator(t *testing.T) {
 	require.Error(t, err)
 	require.Nil(t, iter3)
 }
+
+func TestParallelWrites(t *testing.T) {
+	db, err := New(t.TempDir())
+	require.NoError(t, err)
+	defer db.Close()
+
+	wg := sync.WaitGroup{}
+
+	// start 10 goroutines that write to the database
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		go func(i int) {
+			t.Log("start time", i, time.Now())
+			defer wg.Done()
+			cs := new(store.Changeset)
+			for j := 0; j < 100; j++ {
+				key := fmt.Sprintf("key-%d-%03d", i, j)
+				val := fmt.Sprintf("val-%d-%03d", i, j)
+
+				cs.AddKVPair(store.KVPair{StoreKey: storeKey1, Key: []byte(key), Value: []byte(val)})
+			}
+
+			require.NoError(t, db.ApplyChangeset(uint64(i+1), cs))
+			t.Log("end time", i, time.Now())
+		}(i)
+
+	}
+
+	wg.Wait()
+}
+
+func TestParallelWriteAndPruning(t *testing.T) {
+	db, err := New(t.TempDir())
+	require.NoError(t, err)
+	defer db.Close()
+
+	latestVersion := 100
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
```

Comment:
Please immediately invoke `defer wg.Wait()` after this line instead of down below.